gobblin-metastore/src/main/resources/db/migration/V1_0_1__Add_schema_modification_sprocs.sql

```diff
@@ -0,0 +1,113 @@
+-- Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+-- this file except in compliance with the License. You may obtain a copy of the
+-- License at  http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software distributed
+-- under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+-- CONDITIONS OF ANY KIND, either express or implied.
+
+--
+-- Version: 1.0.1
+--
+-- Helper stored procedures that enable idempotent schema modifications in MySql.
+-- The following stored procedures are being added:
+--   1. add_index_if_not_exists: check if the specified index exists on the
+--                                  given table and create it if it does not.
+--   2. drop_index_if_exists: check if the specified index exists on the
+--                            given table and delete it if it does.
+--   3. add_column_if_not_exists: check if the specified columns exists on the
+--                                give table and create it if it does not.
+--   4. modify_column_if_exists: check if the specified column exists on the
+--                               given table and modifies it if it does.
+--
+
+DELIMITER //
```

Comment:
Actually, one maybe dumb question -- if we know the initial state of the database (from the initial schema & migration), and we control all schema additions/deletions, shouldn't these functions be unnecessary?